engine/job.go

```diff
@@ -69,32 +63,38 @@ func (job *Job) Run() error {
 	if job.Eng.Logging {
 		logrus.Infof("+job %s", job.CallString())
 		defer func() {
-			// what if err is nil?
-			logrus.Infof("-job %s%s", job.CallString(), job.err)
+			okerr := "OK"
+			if err != nil {
+				okerr = fmt.Sprintf("ERR: %s", err)
+			}
+			logrus.Infof("-job %s %s", job.CallString(), okerr)
 		}()
 	}
 	var errorMessage = bytes.NewBuffer(nil)
 	job.Stderr.Add(errorMessage)
 	if job.handler == nil {
-		job.err = fmt.Errorf("%s: command not found", job.Name)
-	} else {
-		job.err = job.handler(job)
-		job.end = time.Now()
+		return fmt.Errorf("%s: command not found", job.Name)
 	}
-	if job.closeIO {
+
+	defer func() {
```

Comment:
mmm just a thought, job's stdout/err/in are opened in engine/engine.go and as of how master is now they will never be closed (assuming closeIO is true, which is the default) if job.Run routine didn't reach at least the job = job.handler(job) 
There are 2 return before handling the job, am I correct? @LK4D4